src/registries/npm-registry.js

```diff
@@ -347,7 +346,24 @@ export default class NpmRegistry extends Registry {
   }
 
   getScopedOption(scope: string, option: string): mixed {
-    return this.getOption(scope + (scope ? ':' : '') + option);
+    const key = scope + (scope ? ':' : '') + option;
+
+    let val = this.getOption(key);
+
+    if (typeof val === 'undefined') {
+      const registries = Object.keys(this.registries);
+      for (const registryName of registries) {
+        if (this.registries[registryName].getOption) {
+          val = this.registries[registryName].getOption(key);
+        }
+
+        if (typeof val !== 'undefined') {
+          break;
+        }
+      }
+    }
+
+    return val;
```

Comment:
Do you think it will be possible to swap this for an implementation of `YarnRegistry.getScopedOption`?

Reply:
I believe that would turn into endless recursion.

`NpmRegistry.getScopedOption` would loop through all registries looking for `getScopedOption` and call it, however if it doesn't exist in that registry, it in turn would do the same thing.

Reply:
Sorry, I'll clarify my meaning. How aboutif we implement `YarnRegistry.getScopedOption` as something like:
```javascript
  getScopedOption(scope: string, option: string): mixed {
    return this.getOption(scope + (scope ? ':' : '') + option);
}
```

Reply:
Technically, it already implements `getScopedOption` as it is an extension of `NpmRegistry`.  I forgot about this fact; I have to undo one of my changes.

I could look into moving all the logic into `BaseRegistry` and remove them from both `YarnRegistry` and `NpmRegistry` if that's the desired goal?

Reply:
I think we can start by adding a test for this and then playing around to see what works best. It'll be much easier that way. Makes sense?